rfcs/0000-support-injected-sidecars.md

```diff
@@ -0,0 +1,124 @@
+3. We will put this new behavior behind a feature gate (Environment variable added to kpack controller deployment
+   called `SUPPORT_INJECTED_SIDECARS`) and only execute in regular containers if enabled by the user. In
+   the event that the feature flag is not set or set to false, kpack will continue to execute builds in init containers.
+4. Due to the fact that sidecars such as istio-proxy, do not exit when all other containers exit, we will need to add a
+   method for stopping the proxy once a build exits
+    1. To do this we can update the running pod's `spec.activeDeadlineSeconds` to 1 which will cause the pod to go into
+       a failed state and kill all containers after 1 second.
+    2. This will kill the sidecar, but it will also leave our pod in a failed state with the message `DeadlineExceeded`,
+       leading the reconciler to believe the build has failed. To circumvent this we will need allow pods to
+       fail with the reason `DeadlineExceeded` as long the completion container has successfully completed.
+5. We will also need a way to delay the start of a build until the sidecars are ready in the event that they are
+   required for network communication  (i.e istio)
+    1. To do this we will have the prepare step wait on a file passed in through
+       the [kubernetes downward api][downward-api]. This file can be provided from a `build ready` annotation that is
+       added to the pod when by the build reconciler when all the containers in the pod are in a ready state.
+
+[kpack-spike-build-waiter]: https://github.com/pivotal/kpack/blob/execute-builds-in-regular-containers/cmd/build-waiter/main.go
+
+[step-modifier]: https://github.com/pivotal/kpack/blob/44257a70c2c9e9ac5703eadc871c7c9be22cfadc/pkg/apis/build/v1alpha2/build_pod.go#L181
+
+[downward-api]: https://kubernetes.io/docs/concepts/workloads/pods/downward-api/
+
+## Complexity
+
+* Our current build logging relies heavily on the fact that our builds exist in init containers so that logic will
+  need to be reworked to support this change.
+* We would need to create an implementation specific to windows builds as well if windows support is needed
+
+## Prior Art
+
+### 1. tekton
+
+https://github.com/tektoncd/pipeline/tree/main/cmd/entrypoint#waiting-for-sidecars
+
+### 2. kpack spike
+
+https://github.com/pivotal/kpack/pull/1019
+
+## Alternatives
+
+* We can create tekton [`TaskRuns`][tekton-task-run] to execute each step of the build since tekton has come up with
+  their own way to solve this problem which
+  involves [replacing the sidecar container images with a "noop" image][tekton-stop-sidecar].
+    * Note: Tekton currently will only stop sidecars (including the istio proxy) if at least one sidecar is defined in
+      the `TaskSpec`.
+* We can build the Tekton entrypoint code as an image to use in kpack.
+* We could execute all build steps in one standard container, similar to [Pack's][pack-build-docs] `pack build` command
```

Comment:
Don't you think it's worth doing that?

Reply:
Which one are you referring to, the github comment diff is showing multiple lines for me so I can't tell which one

Reply:
oops, I thought I marked different lines. Sorry about that.
I refer to:

> We could execute all build steps in one standard container, similar to [Pack's][pack-build-docs] `pack build` command